libs/ngxtension/inject-local-storage/src/inject-local-storage.ts

```diff
@@ -124,27 +123,74 @@ const internalInjectLocalStorage = <R>(
 		const localStorage = inject(NGXTENSION_LOCAL_STORAGE);
 		const destroyRef = inject(DestroyRef);
 
-		const initialStoredValue = goodTry(() => localStorage.getItem(key));
-		const initialValue = initialStoredValue
-			? (goodTry(() => parse(initialStoredValue) as R) ?? defaultValue)
-			: defaultValue;
-		const internalSignal = signal(initialValue);
-
-		effect(() => {
-			const value = internalSignal();
-			if (value === undefined) {
-				goodTry(() => localStorage.removeItem(key));
-			} else {
-				goodTry(() => localStorage.setItem(key, stringify(value)));
+		const initialStoredValue = goodTry(() => localStorage.getItem(key), null);
+		const internalSignal = signal<R>(
+			initialStoredValue
+				? goodTry(() => parse(initialStoredValue) as R, defaultValue)
+				: defaultValue,
+		);
+
+		function syncValueWithLocalStorage(value: R): void {
+			const newValue = goodTry(
+				() => (value === undefined ? null : stringify(value)),
+				null,
+			);
+
+			try {
+				if (newValue === localStorage.getItem(key)) {
+					return;
+				}
+
+				if (newValue === null) {
+					localStorage.removeItem(key);
+				} else {
+					localStorage.setItem(key, newValue);
+				}
+
+				// We notify other consumers in this tab about changing the value in the store for synchronization
+				window.dispatchEvent(
```

Comment:
nit: use DI for `window`

Reply:
use the DOCUMENT token to access the window object

Reply:
fixed

Reply:
fixed